### PR TITLE
Implementing copying feature (parser)

### LINKED ISSUE #1: Add something to copy a directory's items recursively
Create a function (With documentation) to recursively look through a given directory, and output a std::unordered_map<std::filesystem::path, std::vector<file>> that models the directory. 

```txt
TemplateDir
- build 
- src
  - main.cpp -> Contains "int main() {}"
  - LICENSE.txt -> Contains "Input license here. Don't know why it's in the src dir" 
- makefile -> Empty 
``` 

Upon running 

```ps
cinitpp <PATH_TO_TEMPLATE_DIR>/TemplateDir/
```

Returns the equivalent of
```c++
std::unordered_map<std::filesystem::path, std::vector<file>> {
    {".", {file("makefile")}}, 
    {"build", {}}, 
    {"src", {file("main.cpp", "int main() {}"), 
             file("LICENSE.txt", "Input license here. Don't know why it's in the src dir"
    }
}
```

### PR DESCRIPTION
**Pull request by ErichToaster**
# TL;DR

Implemented the copying feature

## Description of changes

I implemented the copying feature in form of a directory parser and an arg parser CLI. The usage printing has been updated in accordance.

I have also updated the standard a tiny bit so that the stored format no longer contains "./" at the beginning of each path, as this would explicitly point to the current directory. Instead, this explicit distinction is added when it is needed, at init time.

## Checklist
- [x] Added description of change
- [x] Relevant documentation/comments is changed or added
- [x] I acknowledge that all my contributions will be made under the project's license.

```md
For: #1
```

### CODE REVIEW

File: src/main.cc [hunk 1]
```diff
@@ -56,6 +70,8 @@ int main(int argc, char** argv) {
 int Init(std::vector<directoryItem>& directoryStructure) {
   for (auto it = directoryStructure.begin(); it != directoryStructure.end();
        ++it) {
+    it->name = std::filesystem::current_path() / it->name;
```
Comment (Renjian-buchai) [inline src/main.cc:73]: I think this is pointless, why?

**Review by Renjian-buchai — APPROVED** [review]
